snuba/query/validation/signature.py

```diff
@@ -104,6 +113,35 @@ def validate(self, expression: Expression, schema: ColumnSet) -> None:
             )
 
 
+class Literal(ParamType):
+    """
+    Validates that the type of a Literal expression is in a set of
+    allowed types.
+
+    If the expression provided is not a Literal, it accepts it.
+    We may consider later whether we want to enforce only literal
+    expressions can be passed as arguments in certain functions.
+    """
+
+    def __init__(self, types: Set[OptionalScalarType], allow_nullable=True) -> None:
+        self.__valid_types = types
+        if allow_nullable:
+            self.__valid_types.add(type(None))
```

Comment:
I don't think this should type check. Set[OptionalScalarType] is a set of instances of any of OptionalScalarTypes, while type(None) is not in that set. Could you check what mypy says about this ?